pendulum_control/include/pendulum_control/rtt_kpsr_executor.hpp

```diff
@@ -24,32 +10,31 @@
 #include "rttest/rttest.h"
 #include "rttest/utils.hpp"
 
-#include "rmw/rmw.h"
-
-#include "rclcpp/executor.hpp"
-#include "rclcpp/macros.hpp"
-#include "rclcpp/memory_strategies.hpp"
+#include <kpsr_ros2_executor/executor_factory.hpp>
+#include <kpsr_ros2_executor/streaming_executor.hpp>
 
 namespace pendulum_control
 {
-/// Instrumented executor that syncs Executor::spin functions with rttest_spin.
-class RttExecutor : public rclcpp::Executor
+class RttKpsrExecutor
 {
 public:
   /// Constructor
   /**
    * Extends default Executor constructor
    */
-  RttExecutor(
+  RttKpsrExecutor(
     const rclcpp::ExecutorOptions & options = rclcpp::ExecutorOptions())
-  : rclcpp::Executor(options), running(false)
+  : running(false)
+  , kpsrExecutor(kpsr::ros2::ExecutorFactory::createExecutor(false, options))
```

Comment:
Do not create the executor here, but pass it as constructor argument, like the real decorator pattern:
https://sourcemaking.com/design_patterns/decorator

Reply:
Done